scripts/west_commands/runners/stlink_gdbserver.py

```diff
@@ -110,10 +110,23 @@ def do_add_parser(cls, parser: argparse.ArgumentParser):
             help="Port number for GDB client",
         )
 
+        # Additional arguments
+        parser.add_argument(
+            "--no-load",
+            action='store_true',
+            help="Don't flash binary before debug (no GDB 'load' command)",
+        )
```

Comment:
Turn this into a `--load/--no-load` option, consistent with `--rebuild/--no-rebuild` and others. Similar to `--swd` in https://github.com/zephyrproject-rtos/zephyr/pull/98670

```python
        # Additional arguments
        parser.add_argument(
            "--load",
            default=True,
            action=argparse.BooleanOptionalAction,
            help="Optionally flash binary before debug (GDB 'load' command)",
        )
```

This is to make the arguments more consistent and let the last passed argument win.

I also agree with @marc-hb, it would be nice to turn this into a runner capability with some common handling.